target/start-mailserver.sh

```diff
@@ -1042,119 +1042,131 @@ function _setup_ssl() {
 
 	# SSL certificate Configuration
 	case $SSL_TYPE in
-		"letsencrypt" )
-			# letsencrypt folders and files mounted in /etc/letsencrypt
-			if [ -e "/etc/letsencrypt/live/$HOSTNAME/fullchain.pem" ]; then
-				KEY=""
-				if [ -e "/etc/letsencrypt/live/$HOSTNAME/privkey.pem" ]; then
-					KEY="privkey"
-				elif [ -e "/etc/letsencrypt/live/$HOSTNAME/key.pem" ]; then
-					KEY="key"
+	    "letsencrypt" )
+            notify 'inf' "Configuring SSL using 'letsecnrypt'"
+            # letsencrypt folders and files mounted in /etc/letsencrypt
+            local LETSENCRYPT_DOMAIN=""
+            local LETSENCRYPT_KEY=""
+			
+            # first determine the letsencrypt domain by checking both the full hostname or just the domainname if a SAN is used in the cert
+            if [ -e "/etc/letsencrypt/live/$HOSTNAME/fullchain.pem" ]; then
+                LETSENCRYPT_DOMAIN=$HOSTNAME
+            elif [ -e "/etc/letsencrypt/live/$DOMAINNAME/fullchain.pem" ]; then
+                LETSENCRYPT_DOMAIN=$DOMAINNAME
+			else
+				notify 'err' "Cannot access '/etc/letsencrypt/live/"$HOSTNAME"/fullchain.pem' or '/etc/letsencrypt/live/"$DOMAINNAME"/fullchain.pem'"
```

Comment:
Shouldn't the script terminate with an error here? `return 1` or something like that?

Reply:
Currently if SSL encounters an error, it just leaves the default configuration in place.  It does make sense that if you configured SSL, it would be assumed to be running correctly if no errors were actually returned from the function.  A return 1 sounds look a good thing to implement.

Reply:
In addition to the error, return 1 is used after the error.  Upon successfull completion of the letsencrypt portion, return 0 is given.